iam/api-client/src/main/java/iam/snippets/CreateServiceAccountKey.java

```diff
@@ -16,80 +16,36 @@
-    }
-
-    // Construct the service account email.
-    // You can modify the ".iam.gserviceaccount.com" to match the service account name in which
-    // you want to create the key.
-    // See, https://cloud.google.com/iam/docs/creating-managing-service-account-keys?hl=en#creating
-    String serviceAccountEmail = serviceAccountName + "@" + projectId + ".iam.gserviceaccount.com";
-    try {
-      ServiceAccountKey key =
-          service
-              .projects()
-              .serviceAccounts()
-              .keys()
-              .create(
-                  "projects/-/serviceAccounts/" + serviceAccountEmail,
-                  new CreateServiceAccountKeyRequest())
-              .execute();
-
-      // The privateKeyData field contains the base64-encoded service account key
-      // in JSON format.
-      // TODO(Developer): Save the below key (jsonKeyFile) to a secure location.
-      //  You cannot download it later.
-      String jsonKeyFile = new String(Base64.getDecoder().decode(key.getPrivateKeyData()));
-
+  public static String createKey(String projectId, String accountName) {
+    // String projectId = "project-id";
+    // String accountName = "my-service-account-name";
+    String email = accountName + "@" + projectId + ".iam.gserviceaccount.com";
+    try (IAMClient iamClient = IAMClient.create()) {
+      CreateServiceAccountKeyRequest req = createAccountKeyReq(projectId, email);
+      ServiceAccountKey createdKey = iamClient.createServiceAccountKey(req);
       System.out.println("Key created successfully");
-      String keyName = key.getName();
+
+      String keyName = createdKey.getName();
       return keyName.substring(keyName.lastIndexOf("/") + 1).trim();
-    } catch (IOException e) {
-      System.out.println("Unable to create service account key: \n" + e);
+    } catch (IOException ex) {
+      System.out.println("Failed to create key");
       return null;
     }
   }
 
-  private static Iam initService() throws GeneralSecurityException, IOException {
-    // Use the Application Default Credentials strategy for authentication. For more info, see:
-    // https://cloud.google.com/docs/authentication/production#finding_credentials_automatically
-    GoogleCredentials credential =
-        GoogleCredentials.getApplicationDefault()
-            .createScoped(Collections.singleton(IamScopes.CLOUD_PLATFORM));
-    // Initialize the IAM service, which can be used to send requests to the IAM API.
-    Iam service =
-        new Iam.Builder(
-            GoogleNetHttpTransport.newTrustedTransport(),
-            GsonFactory.getDefaultInstance(),
-            new HttpCredentialsAdapter(credential))
-            .setApplicationName("service-account-keys")
+  private static CreateServiceAccountKeyRequest createAccountKeyReq(String projId, String email) {
+    return CreateServiceAccountKeyRequest.newBuilder()
+            .setName("projects/" + projId + "/serviceAccounts/" + email)
             .build();
```

Comment:
Inline this code as part of the `createKey` method. Kindly refrain from using shortforms like 'projId'.